src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts

```diff
@@ -42,4 +44,116 @@ describe('CommentEditFormComponent', () => {
+    expect(component.isVisibilityTableExpanded).toBeFalsy();
+    component.toggleVisibilityTable();
+    // After toggling, it should be expanded
+    expect(component.isVisibilityTableExpanded).toBeTruthy();
+  });
+
+  it('should toggle visibility table from expanded to collapsed', () => {
+    component.isVisibilityTableExpanded = true;
+    expect(component.isVisibilityTableExpanded).toBeTruthy();
+    component.toggleVisibilityTable();
+    // After toggling, it should be collapsed
+    expect(component.isVisibilityTableExpanded).toBeFalsy();
+  });
+
+  // triggerModelChange
+  it('should trigger model change with updated field', () => {
+    const testField = 'commentText';
+    const testData = 'Updated comment text';
+
+    const emitSpy = jest.spyOn(component.modelChange, 'emit');
+
+    component.triggerModelChange(testField, testData);
+
+    // Verify that the model should be updated
+    expect(emitSpy).toHaveBeenCalledWith({
+      ...component.model,
+      [testField]: testData,
+    });
+  });
+
+  // triggerModelChangeBatch
+  it('should emit the updated model when triggerModelChangeBatch is called', () => {
+    component.model = {
+      commentText: 'Initial Comment',
+      isUsingCustomVisibilities: false,
+      showCommentTo: [],
+      showGiverNameTo: [],
+    };
+    fixture.detectChanges();
+
+    const updatedModel = {
+      commentText: 'Updated Comment',
+      isUsingCustomVisibilities: true,
+      showCommentTo: ['Public'],
+      showGiverNameTo: ['Team'],
+    };
+
+    const modelChangeSpy = jest.spyOn(component.modelChange, 'emit');
+
+    component.triggerModelChangeBatch(updatedModel);
+
+    // Check if the model is updated
+    expect(modelChangeSpy).toHaveBeenCalledWith(updatedModel);
+  });
+
+  // triggerCloseCommentBoxEvent
+  it('should emit the closeCommentBoxEvent when triggerCloseCommentBoxEvent is called', () => {
+    const closeCommentBoxEventSpy = jest.spyOn(component.closeCommentBoxEvent, 'emit');
+    component.triggerCloseCommentBoxEvent();
+    // Check if the closeCommentBoxEvent was emitted
+    expect(closeCommentBoxEventSpy).toHaveBeenCalled();
+  });
+
+  // triggerSaveCommentEvent
+  it('should emit the saveCommentEvent when triggerSaveCommentEvent is called', () => {
+    const saveCommentEventSpy = jest.spyOn(component.saveCommentEvent, 'emit');
+    component.triggerSaveCommentEvent();
+    // Check if the saveCommentEvent was emitted
+    expect(saveCommentEventSpy).toHaveBeenCalled();
+  });
+
+  // modifyVisibilityControl
+  it('should allow and disallow visibility control and trigger model change', () => {
+    const isAllowed = true;
+    const visibilityType = CommentVisibilityType.GIVER;
+    const visibilityControl = CommentVisibilityControl.SHOW_COMMENT;
+
+    jest.spyOn(component.visibilityStateMachine, 'allowToSee');
+    jest.spyOn(component.visibilityStateMachine, 'disallowToSee');
+    jest.spyOn(component, 'triggerModelChangeBatch');
```

Comment:
let's keep it consistent with the rest of the spys, have them assigned to a const and use that const for `expect`